Feature/elk

the-man-w-laughs opened the pull request:
Elk is used to collect and analyze microservice logs.

the-man-w-laughs committed:
Refactor appsettings.json

Description:
- Move shared configuration to appsettings.json;
- Move unique configuration to appsettings.Docker.json or appsettings.Development.json.

the-man-w-laughs committed:
Refactor ReportHub

Major changes:
- take into account empty data seed;
- add offset and limit to GetAllAsync in IMongoRepository;
- change identifier names to more appropriate ones;
- add ProjectsInfoController that responsible for ProjectInfo (Now User can retrieve all the projects User have ever generated reports for)
  add the appropriate service IProjectReportInfoService;
- remove unused UpdatedAt field from ProjectReportInfo;
- uppercase namespaces names.

the-man-w-laughs committed:
Add initial elasticsearch configuration

Major changes:
- add elasticsearch and kibana to docker-compose;
- add ElasticConfiguration to appsettings.json;
- add Serilog configuration to appsettings.json;
- add LoggerConfigurationExtensions.

the-man-w-laughs committed:
Add logging

the-man-w-laughs committed:
Merge branch 'feature/SignalR' into feature/elk

the-man-w-laughs committed:
Add logging to Identity services.

the-man-w-laughs committed:
Add logging to ReportHub

Major changes:
- add logging to ReportHub;
- update logging messages in Identity.

the-man-w-laughs committed:
Add logging to TeamHub

Major changes:
- add logging to comment command handlers.

the-man-w-laughs committed:
Final addtion of logs to TeamHub.

the-man-w-laughs committed:
Remove logs from blocks with throw statement.

the-man-w-laughs committed:
Add logging to ExceptionMiddleware.

the-man-w-laughs committed:
Use WebApi SubName to form ElasticSearch index name.

mebry commented on src/Shared/Repository/NoSql/Mongo/IMongoRepository.cs:
```diff
@@ -5,14 +5,13 @@ namespace Shared.Repository.NoSql
-            Expression<Func<TEntity, bool>> filter,
+            Expression<Func<TEntity, bool>> filter = null,
+            int offset = 0,
+            int limit = int.MaxValue,
```
It's better not to fetch all records by default

mebry commented on src/services/ReportHub/ReportHub.BLL/Extensions/FullProjectParserExtensions.cs:
```diff
@@ -13,40 +13,61 @@ public static string ToReport(this FullProjectInfoResponse fullProjectInfo)
-                report.AppendLine($"\t- {teamMember.Email}");
+                report.AppendLine($"\tNo Team Members");
             }
 
             report.AppendLine("Tasks:");
-
-            foreach (var task in fullProjectInfo.Tasks)
+            if (fullProjectInfo.Tasks != null)
```
Empty line is missed here

mebry commented on src/services/ReportHub/ReportHub.WebApi/Controllers/ProjectsInfoController.cs:
```diff
@@ -0,0 +1,38 @@
+    /// <summary>
+    /// Get all awailable user project infos
+    /// </summary>
+    [HttpGet("all")]
+    public async Task<IActionResult> GetAllUserProjectInfos(
```
1)Fix the routing
2)For asynchronous methods, it's necessary to append Async to the end of the method name

the-man-w-laughs committed:
Implement Mentor's Recommendations

Major changes:
- add empty line where necessary;
- make offset and limit required parameters;
- add async to asynchronous methods;
- fix routing for GetAllUserProjectInfosAsync.
